ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/JdbcLedgerDao.scala

```diff
@@ -1377,49 +1434,123 @@ private class JdbcLedgerDao(
             "ledger_offset" -> offset,
             "recorded_at" -> Instant.now(),
             "submission_id" -> entry.submissionId,
-            "participant_id" -> entry.submissionId,
+            "participant_id" -> entry.participantId,
             "typ" -> entry.value,
             "rejection_reason" -> reasonOrNull(entry)
           )
           .execute()
         PersistenceResponse.Ok
       }).recover {
-          case NonFatal(e) if e.getMessage.contains(queries.DUPLICATE_KEY_ERROR) =>
-            logger.warn(
-              s"Ignoring duplicate package upload entry submission for submissionId ${entry.submissionId} participantId ${entry.participantId}")
-            conn.rollback()
-            PersistenceResponse.Duplicate
-        }
-        .getOrElse {
+        case NonFatal(e) if e.getMessage.contains(queries.DUPLICATE_KEY_ERROR) =>
           logger.warn(
-            s"Insert package upload entry failed for submissionId ${entry.submissionId} participantId ${entry.participantId}")
+            s"Ignoring duplicate package upload entry submission for submissionId ${entry.submissionId} participantId ${entry.participantId}")
           conn.rollback()
           PersistenceResponse.Duplicate
-        }
+      }.get
+    }
+  }
+
+  override def getPartyAllocationEntries(
+      startInclusive: Long,
+      endExclusive: Long): Source[(Long, PartyAllocationLedgerEntry), NotUsed] =
+    paginatingStream(
+      startInclusive,
+      endExclusive,
+      PageSize,
+      (startI, endE) => {
+        dbDispatcher
+          .executeSql(s"load party allocation entries", Some(s"bounds: [$startI, $endE[")) {
+            implicit conn =>
+              SQL_SELECT_PARTY_ALLOCATION_ENTRIES
+                .on("startInclusive" -> startI, "endExclusive" -> endE)
+                .as(partyAllocationEntryParser.*)
+          }
+      }
+    ).flatMapConcat(Source(_))
+
+  override def lookupPartyAllocationEntry(
+      submissionId: SubmissionId): Future[Option[PartyAllocationLedgerEntry]] = {
+    dbDispatcher
+      .executeSql(s"load party allocation entry", None) { implicit conn =>
+        SQL_SELECT_PARTY_ALLOCATION_ENTRY
+          .on("submissionId" -> submissionId)
+          .as(partyAllocationEntryParser.*)
+          .headOption
+          .map(_._2)
+      }
+  }
+
+  override def storePartyAllocationEntry(
+      offset: LedgerOffset,
+      newLedgerEnd: LedgerOffset,
+      externalOffset: Option[ExternalOffset],
+      submissionId: SubmissionId,
+      participantId: ParticipantId,
+      entry: PartyAllocationLedgerEntry): Future[PersistenceResponse] = {
+
+    dbDispatcher.executeSql("store party allocation reject entry") { implicit conn =>
+      updateLedgerEnd(newLedgerEnd, externalOffset)
+      Try({
+        SQL(queries.SQL_INSERT_PARTY_ALLOCATION_ENTRY)
+          .on(
+            "ledger_offset" -> offset,
+            "recorded_at" -> Instant.now(),
```

Comment:
This needs to use `recordTime` from the allocation entries.

Reply:
now corrected